app/Commands/Sites/CreateCommand.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace App\Commands\Sites;
+
+use App\Commands\BaseCommand;
+use App\Commands\Concerns\HasServerIdParameter;
+use App\Helpers\OptionsHelper;
+use App\Questions\Ask;
+use App\Questions\Choice;
+use App\Questions\Confirm;
+use App\Questions\HasQuestions;
+use Illuminate\Support\Str;
+
+class CreateCommand extends BaseCommand
+{
+    use HasQuestions;
+    use HasServerIdParameter;
+
+    protected $signature = 'sites:create
+                            {installation_method : Type of installation (wp or blank)}
+                            {server_id? : Server ID}
+                            {--domain= : Domain name}
+                            {--site_user=}
+                            {--db_name=}
+                            {--db_user=}
+                            {--db_pass=}
+                            {--wp_title=}
+                            {--wp_admin_user=}
+                            {--wp_admin_email=}
+                            {--wp_admin_pass=}
+                            {--php_version=}
+                            {--page_cache_enabled}
+                            {--https_enabled}
+                            {--profile=}
+                            {--f|force}';
+
+    protected $description = 'Create a site';
+
+    protected array $userInput;
+
+    protected function action(): int
+    {
+        if (!in_array($this->argument('installation_method'), OptionsHelper::INSTALLATION_METHODS, true)) {
+            $this->error('Invalid site type.');
+            $this->newLine(1);
+            return self::INVALID;
+        }
```

Comment:
Why are we not making this option interactive too?

Reply:
Because it's an argument not an option. 

Arguments have a specific order in the command signature, so the first value given as an argument is assigned to the first named argument. Meaning given `spinupwp site:create 15 --domain=example.com...` then `$this->argument('installation_method') == 15`. 

We could add logic to check `is_numeric($this->argument('installation_method'))` but that would sort of defeat the purpose of named arguments. 

Alternately, we change the command signature and make `installation_method` an option instead of an argument.

Another option would be to add `installation_method` as both an argument **and** an option but this seems the least optimal.

On the other hand, if **all** arguments are missing we could interactively prompt. It's only an issue when _some_ fo the arguments are missing.

Thoughts?

Reply:
I think it makes sense to always prompt for everything that hasn't been given an answer via a flag once we encounter any single thing we need to prompt for. 

We basically treat it as a sign that the user might not be thinking in terms of the definitive list of fields, so let's verify everything and help them get this how they want it now rather than need to tweak it later. 

I'd much rather hit enter on default values for a couple of extra prompts than overlook flags and potentially not realize the difference at all.

Reply:
> Alternately, we change the command signature and make installation_method an option instead of an argument.

This seems a better approach and prompt when not supplied.

Reply:
Done